csharp/ql/src/experimental/Security Features/JsonWebTokenHandler/JsonWebTokenHandlerLib.qll

```diff
@@ -0,0 +1,263 @@
+import csharp
+import DataFlow
+
+/**
+ * A sensitive property for `TokenValidationParameters` that updates the underlying value.
+ */
+class TokenValidationParametersPropertySensitiveValidation extends Property {
+  TokenValidationParametersPropertySensitiveValidation() {
+    exists(Class c |
+      c.hasQualifiedName("Microsoft.IdentityModel.Tokens.TokenValidationParameters")
+    |
+      c.getAProperty() = this and
+      this.getName() in [
+          "ValidateIssuer", "ValidateAudience", "ValidateLifetime", "RequireExpirationTime",
+          "RequireAudience"
+        ]
+    )
+  }
+}
+
+/**
+ * A dataflow from a `false` value to a write sensitive property for `TokenValidationParameters`.
+ */
+class FalseValueFlowsToTokenValidationParametersPropertyWriteToBypassValidation extends DataFlow::Configuration {
+  FalseValueFlowsToTokenValidationParametersPropertyWriteToBypassValidation() {
+    this = "FlowsToTokenValidationResultIsValidCall"
+  }
+
+  override predicate isSource(DataFlow::Node source) {
+    source.asExpr().getValue() = "false" and
+    source.asExpr().getType() instanceof BoolType
+  }
+
+  override predicate isSink(DataFlow::Node sink) {
+    sink.asExpr() = any(TokenValidationParametersPropertySensitiveValidation p).getAnAssignedValue()
+  }
+}
+
+/**
+ * Holds if `assemblyName` is older than version `ver`
+ */
+bindingset[ver]
+predicate isAssemblyOlderVersion(string assemblyName, string ver) {
+  exists(Assembly a |
+    a.getName() = assemblyName and
+    a.getVersion().isEarlierThan(ver)
+  )
+}
+
+/**
+ * A method `ValidateToken` for `Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler` or other Token handler that shares the same behavior characteristics
+ */
+class JsonWebTokenHandlerValidateTokenMethod extends Method {
+  JsonWebTokenHandlerValidateTokenMethod() {
+    this.hasQualifiedName("Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateToken") or
+    this.hasQualifiedName("Microsoft.AzureAD.DeviceIdentification.Common.Tokens.JwtValidator.ValidateEncryptedToken")
+    //// TODO: ValidateEncryptedToken has the same behavior than ValidateToken, but it will be changed in a future release
+    ////       The line below would allow to check if the ValidateEncryptedToken version used meets the minimum requirement
+    ////       Once we have the fixed assembly version we can uncomment the line below
+    // and isAssemblyOlderVersion("Microsoft.AzureAD.DeviceIdentification", "0.0.0")
```

Comment:
Is this assembly version known now?

Reply:
Good catch, I forgot to check that one... I will investigate